query-algebrizer/src/clauses/fulltext.rs

```diff
@@ -102,15 +95,27 @@ impl ConjoiningClauses {
         // term before the attribute arguments and collect the (variadic) attributes into a set.
         // let a: Entid  = self.resolve_attribute_argument(&where_fn.operator, 1, args.next().unwrap())?;
         //
-        // TODO: allow non-constant attributes.
         // TODO: improve the expression of this matching, possibly by using attribute_for_* uniformly.
         let a = match args.next().unwrap() {
             FnArg::IdentOrKeyword(i) => schema.get_entid(&i),
             // Must be an entid.
             FnArg::EntidOrInteger(e) => Some(e),
+            FnArg::Variable(v) => {
+                // If it's already bound, then let's expand the variable.
+                self.bound_value(&v).and_then(|tv| {
+                    if let TypedValue::Ref(entid) = tv {
+                        Some(entid)
+                    } else {
```

Comment:
You might want to differentiate a non-`ValueType::Ref` from the errors below; your call.